bug: panic: runtime error: invalid memory address or nil pointer dereference

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x192d87d]

goroutine 104 [running]:
github.com/streamnative/oxia/coordinator/impl.(*shardController).internalNewTermAndAddFollower(0xc0001d0b00, {0x2534a60?, 0xc000af2730?}, {{0xc00011dea0, 0x45}, {0xc000516cc0, 0x2f}}, 0xc0005d73b0)
	/src/oxia/coordinator/impl/shard_controller.go:513 +0xfd
github.com/streamnative/oxia/coordinator/impl.(*shardController).run(0xc0001d0b00)
	/src/oxia/coordinator/impl/shard_controller.go:202 +0x545
github.com/streamnative/oxia/common.DoWithLabels.func1({0x2534a28?, 0xc00077f890?})
	/src/oxia/common/pprof.go:46 +0x13
runtime/pprof.Do({0x2534a60?, 0xc000139950?}, {{0xc00068eb40?, 0x0?, 0x0?}}, 0xc000484f30)
	/usr/local/go/src/runtime/pprof/runtime.go:51 +0x8c
github.com/streamnative/oxia/common.DoWithLabels({0x2534a60, 0xc000139950}, 0xc000644210, 0xc00063e1d0)
	/src/oxia/common/pprof.go:42 +0x34d
created by github.com/streamnative/oxia/coordinator/impl.NewShardController in goroutine 1
	/src/oxia/coordinator/impl/shard_controller.go:150 +0xba5
```

--- a/common/error_codes.go
+++ b/common/error_codes.go
@@ -33,7 +33,6 @@ const (
 	CodeNamespaceNotFound       codes.Code = 110
 	CodeNotificationsNotEnabled codes.Code = 111
 	CodeFollowerAlreadyPresent  codes.Code = 112
-	CodeFollowerAlreadyFenced   codes.Code = 113
 )
 
 var (
@@ -50,5 +49,4 @@ var (
 	ErrorNamespaceNotFound       = status.Error(CodeNamespaceNotFound, "oxia: namespace not found")
 	ErrorNotificationsNotEnabled = status.Error(CodeNotificationsNotEnabled, "oxia: notifications not enabled on namespace")
 	ErrorFollowerAlreadyPresent  = status.Error(CodeFollowerAlreadyPresent, "oxia: follower is already present")
-	ErrorFollowerAlreadyFenced   = status.Error(CodeFollowerAlreadyFenced, "oxia: follower is already fenced")
 )

--- a/coordinator/impl/shard_controller.go
+++ b/coordinator/impl/shard_controller.go
@@ -498,7 +498,7 @@ func (s *shardController) newTermAndAddFollower(ctx context.Context, node model.
 
 func (s *shardController) internalNewTermAndAddFollower(ctx context.Context, node model.ServerAddress, res chan error) {
 	fr, err := s.newTerm(ctx, node)
-	if err != nil && status.Code(err) != common.CodeFollowerAlreadyFenced {
+	if err != nil {
 		res <- err
 		return
 	}

--- a/server/follower_controller.go
+++ b/server/follower_controller.go
@@ -267,16 +267,6 @@ func (fc *followerController) NewTerm(req *proto.NewTermRequest) (*proto.NewTerm
 			slog.Int64("new-term", req.Term),
 		)
 		return nil, common.ErrorInvalidTerm
-	} else if req.Term == fc.term && fc.status != proto.ServingStatus_FENCED {
-		// It's OK to receive a duplicate Fence request, for the same term, as long as we haven't moved
-		// out of the Fenced state for that term
-		fc.log.Warn(
-			"Failed to fence with same term in invalid state",
-			slog.Int64("follower-term", fc.term),
-			slog.Int64("new-term", req.Term),
-			slog.Any("status", fc.status),
-		)
-		return nil, common.ErrorFollowerAlreadyFenced
 	}
 
 	if fc.db == nil {

--- a/server/follower_controller_test.go
+++ b/server/follower_controller_test.go
@@ -368,8 +368,7 @@ func TestFollower_DuplicateNewTermInFollowerState(t *testing.T) {
 
 	stream := newMockServerReplicateStream()
 	go func() {
-		// cancelled due to fc.Close() below
-		assert.ErrorIs(t, fc.Replicate(stream), context.Canceled)
+		assert.NoError(t, fc.Replicate(stream))
 	}()
 
 	stream.AddRequest(createAddRequest(t, 1, 0, map[string]string{"a": "0", "b": "1"}, 10))
@@ -384,8 +383,21 @@ func TestFollower_DuplicateNewTermInFollowerState(t *testing.T) {
 	}, 10*time.Second, 10*time.Millisecond)
 
 	r, err := fc.NewTerm(&proto.NewTermRequest{Term: 1})
-	assert.Nil(t, r)
-	assert.Equal(t, common.CodeFollowerAlreadyFenced, status.Code(err))
+	assert.NoError(t, err)
+	assert.NotNil(t, r)
+	assert.EqualValues(t, r1.Offset, r.HeadEntryId.Offset)
+	assert.EqualValues(t, 1, r.HeadEntryId.Term)
+
+	stream.AddRequest(createAddRequest(t, 1, 1, map[string]string{"a": "1", "b": "2"}, 11))
+
+	// Wait for acks
+	r2 := stream.GetResponse()
+
+	assert.EqualValues(t, 1, r2.Offset)
+
+	assert.Eventually(t, func() bool {
+		return fc.CommitOffset() == 1
+	}, 10*time.Second, 10*time.Millisecond)
 
 	assert.NoError(t, fc.Close())
 	assert.NoError(t, kvFactory.Close())

